plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -419,17 +433,45 @@ private boolean validateQueryFieldInQueryString(
                 }
             }
         }
-
         return true;
+    }
+
+    /**
+     * Validates that the query fields specified in the input and output mappings exist in the query string.
+     * @param processInputMap  the list of input mappings
+     * @param processOutputMap the list of output mappings
+     * @param queryString      the query string to be validated
+     * @return true if all query fields exist in the query string, false otherwise
+     */
+    private boolean validateQueryFieldInQueryString(
+        List<Map<String, String>> processInputMap,
+        List<Map<String, String>> processOutputMap,
+        String queryString
+    ) {
+        boolean isInputValid;
+        boolean isOutputValid;
+        if (processInputMap != null && processInputMap.size() > 0) {
+            isInputValid = validateRequiredInputMappingFields(processInputMap, queryString);
+        } else {
+            isInputValid = true;
+        }
+
```

Comment:
We should check if the `isInputValid` is false here to early stop when we want to return `isInputValid && isOutputValid` in the end.

Reply:
added 
```
        if(!isInputValid) {
            return false;
        }
```